runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java

```diff
@@ -236,47 +201,55 @@ void update()
 					panel.revalidate();
 				}
 
-				levelLabel.setText(String.valueOf(xpInfo.getLevel()));
-				xpGained.setText(XpPanel.formatLine(xpInfo.getXpGained(), "xp gained"));
-				xpLeft.setText(XpPanel.formatLine(xpInfo.getXpRemaining(), "xp left"));
-				actionsLeft.setText(XpPanel.formatLine(xpInfo.getActionsRemaining(), "actions left"));
+				// Update information labels
+				expGained.setText(htmlLabel("Gained: ", xpInfo.getXpGained()));
+				expLeft.setText(htmlLabel("Left: ", xpInfo.getXpRemaining()));
+				actionsLeft.setText(htmlLabel("Actions: ", xpInfo.getActionsRemaining()));
 
-				final int progress = xpInfo.getSkillProgress();
-
-				progressBar.setValue(progress);
-				progressBar.setBackground(Color.getHSBColor((progress / 100.f) * (120.f / 360.f), 1, 1));
+				// Update progress bar
+				progressBar.setValue(xpInfo.getSkillProgress());
+				progressBar.setCenterLabel(xpInfo.getSkillProgress() + "%");
+				progressBar.setLeftLabel("Lvl. " + xpInfo.getLevel());
+				progressBar.setRightLabel("Lvl. " + (xpInfo.getLevel() + 1));
 
 				progressBar.setToolTipText("<html>"
-					+ XpPanel.formatLine(xpInfo.getActions(), "actions")
+					+ xpInfo.getActions() + " actions"
 					+ "<br/>"
-					+ XpPanel.formatLine(xpInfo.getActionsHr(), "actions/hr")
+					+ xpInfo.getActionsHr() + " actions/hr"
 					+ "<br/>"
 					+ xpInfo.getTimeTillLevel() + " till next lvl"
 					+ "</html>");
+
+				progressBar.repaint();
+
 			}
 
-			// Always update xp/hr as time always changes
-			xpHr.setText(XpPanel.formatLine(xpInfo.getXpHr(), "xp/hr"));
+			// Update exp per hour seperately, everytime (not only when there's an update)
+			expHour.setText(htmlLabel("Per hour: ", xpInfo.getXpHr()));
 		});
 	}
 
-	private static BufferedImage createHoverImage(BufferedImage image)
+	public static String htmlLabel(String key, int value)
 	{
-		LookupTable lookup = new LookupTable(0, 4)
+		String valueStr = value + "";
+
+		if (value > 9999999 || value < -9999999)
 		{
-			@Override
-			public int[] lookupPixel(int[] src, int[] dest)
+			valueStr = "Lots!";
+		}
+		else if (value > 1000)
+		{
+			if (value % 10000 == 0)
 			{
-				if (dest[3] != 0)
-				{
-					dest[3] = 60;
-				}
-
-				return dest;
+				valueStr = value / 1000 + "K";
 			}
-		};
 
-		LookupOp op = new LookupOp(lookup, new RenderingHints(null));
-		return op.filter(image, null);
+			valueStr = NumberFormat.getNumberInstance(Locale.US).format(value)
```

Comment:
Same as before. Why not use the StackFormatter? If there is something specific about not using, it may be worth adding to it or creating a similar class for skills.